tiledb/sm/rest/rest_client.cc

```diff
@@ -57,6 +57,7 @@
 #include <cassert>
```

Comment:
The changes to rest_client.cc and rest_client.h should be close if not identical to #4696, since I ran into a similar need to have a MemoryTracker on the same deserialization methods. The difference would be that I passed the trackers along to `domain_from_capnp` and `dimension_label_from_capnp` for calls to `make_shared` with Domain objects.